web/packages/teleport/src/Bots/Edit/EditDialog.tsx

```diff
@@ -0,0 +1,298 @@
+                />
+                <FieldSelectAsync
+                  menuPosition="fixed"
+                  label="Roles"
+                  rule={requiredField('At least one role is required')}
+                  placeholder="Click to select roles"
+                  isSearchable
+                  isMulti
+                  isClearable={false}
+                  value={(selectedRoles ?? data?.roles.toSorted() ?? []).map(
+                    r => ({
+                      value: r,
+                      label: r,
+                    })
+                  )}
+                  onChange={(values: Option[]) =>
+                    setSelectedRoles(values?.map(v => v.value) || [])
+                  }
+                  loadOptions={async input => {
+                    const flags = ctx.getFeatureFlags();
+                    const roles = await fetchRoles({ search: input, flags });
+                    return roles.items.map(r => ({
+                      value: r.name,
+                      label: r.name,
+                    }));
+                  }}
+                  noOptionsMessage={() => 'No roles found'}
+                  elevated={true}
+                />
+                <Box mb={3}>
+                  <TraitsEditor
+                    configuredTraits={
+                      selectedTraits ??
+                      data?.traits
+                        .toSorted((a, b) => a.name.localeCompare(b.name))
+                        .map(t => ({
+                          traitKey: {
+                            value: t.name,
+                            label: t.name,
+                          },
+                          traitValues: t.values.toSorted().map(v => ({
+                            value: v,
+                            label: v,
+                          })),
+                        })) ??
+                      []
+                    }
+                    setConfiguredTraits={setSelectedTraits}
+                    isLoading={false}
+                    label="Traits"
+                    addActionLabel="Add trait"
+                    addActionSubsequentLabel="Add another trait"
+                    autoFocus={false}
+                  />
+                </Box>
+                <FieldInput
+                  label="Max session duration"
+                  rule={requiredField('Max session duration is required')}
+                  value={
+                    selectedMaxSessionDuration ??
+                    formatDuration({
+                      seconds:
+                        data?.max_session_ttl?.seconds ??
+                        TWELVE_HOURS_IN_SECONDS,
+                    })
+                  }
+                  onChange={e => setSelectedMaxSessionDuration(e.target.value)}
+                  helperText={
+                    'A duration string such as 12h, 2h 45m, 43200s. Maximum is 24h. Valid time units are ns, us (or µs), ms, s, m, h.'
```

Comment:
I tried setting ns, us, and ms, but they don't appear to work.
We can probably remove them, I don't think anyone needs a max session duration shorter than one second.